### PR TITLE
feat: Add Input and InputSearch components

### PR DESCRIPTION
**Pull request by knajjars**
https://github.com/user-attachments/assets/ae371daf-bc26-46a7-8c9b-e1b96b0f82da

**AI Description**

This PR updates the version of `react-hook-form` to `^7.52.1` and introduces a new `InputSearch` component.

- The `react-hook-form` package is updated to version `^7.52.1` in `package.json` and `package-lock.json`.
- The `InputSearch` component is added to the `src/interfaces/assistants_web/src/components/Shared/InputSearch/InputSearch.tsx` file.
- The `InputSearch` component is imported and used in multiple files: src/interfaces/assistants_web/src/app/test/page.tsx, src/interfaces/assistants_web/src/components/Agents/AgentsList.tsx, src/interfaces/assistants_web/src/components/Agents/DiscoverAgents.tsx, src/interfaces/assistants_web/src/components/ConversationList/ConversationListPanel.tsx, and src/interfaces/assistants_web/src/components/Shared/InputSearch/index.ts.
- The `Input` component in `src/interfaces/assistants_web/src/components/Shared/Input/Input.tsx` is updated to include a new `actionType` option, `'reveal', and the `disabled` prop is removed.
- The `kind` and `theme` props are removed from the `Input` component in `src/interfaces/assistants_web/src/components/Shared/Dropdown/Dropdown.tsx`.
- The `hasError` prop is removed from the `Input` component in `src/interfaces/assistants_web/src/components/AgentForm.tsx`.
- The `stackPosition` and `errorText` props are removed from the `Input` component in `src/interfaces/assistants_web/src/app/(auth)/login/Login.tsx` and `src/interfaces/assistants_web/src/app/(auth)/register/Register.tsx`.
- The `actionType` prop is updated from `'revealable'` to `'reveal'` in `src/interfaces/assistants_web/src/app/(auth)/login/Login.tsx` and `src/interfaces/assistants_web/src/app/(auth)/register/Register.tsx`.
- The color value `100` is updated from `'#292929'` to `'#1A1A1A'` in `src/interfaces/assistants_web/src/themes/cohereTheme.js`.

### CODE REVIEW

File: src/interfaces/assistants_web/src/app/test/page.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
+'use client';
```
Comment (tomtobac) [inline src/interfaces/assistants_web/src/app/test/page.tsx:1]: can we can remove this page, right?

File: src/interfaces/assistants_web/src/components/Conversation/Composer/WebSearchConfigurationModal.tsx [hunk 2]
```diff
@@ -18,13 +17,6 @@ export const WebSearchModal: React.FC<{
     params: { tools },
   } = useParamsStore();
   const [site, setSite] = useState('');
-  const inputRef = useRef<HTMLInputElement>(null);
-
-  useEffect(() => {
-    if (inputRef.current) {
-      inputRef.current.focus();
-    }
-  }, []);
```
Comment (tomtobac) [inline src/interfaces/assistants_web/src/components/Conversation/Composer/WebSearchConfigurationModal.tsx:27]: I think we should keep this

File: src/interfaces/assistants_web/src/components/ShareModal.tsx [hunk 3]
```diff
@@ -96,8 +96,6 @@ export const ShareModal: React.FC<ShareModalProps> = ({ conversationId }) => {
       )}
       <div className="flex flex-col gap-y-2">
         <Input
-          truncate
-          readOnly
```
Comment (tomtobac) [inline src/interfaces/assistants_web/src/components/ShareModal.tsx:100]: we should support read only

File: src/interfaces/assistants_web/src/components/Shared/InputSearch/InputSearch.tsx [hunk 4]
```diff
@@ -0,0 +1,72 @@
+import { Fieldset, Input } from '@headlessui/react';
+import { useState } from 'react';
+
+import { Icon } from '@/components/Shared/Icon';
+import { STYLE_LEVEL_TO_CLASSES } from '@/components/Shared/Text';
+import { cn } from '@/utils';
+
+type Props = {
```
Comment (tomtobac) [inline src/interfaces/assistants_web/src/components/Shared/InputSearch/InputSearch.tsx:8]: we could extend the input field props and avoid adding max length, classname, placeholder, value, etc

File: src/interfaces/assistants_web/src/app/test/page.tsx (hunk 1 above)
Comment (knajjars) [reply src/interfaces/assistants_web/src/app/test/page.tsx:1]: oh yes!

File: src/interfaces/assistants_web/src/components/Conversation/Composer/WebSearchConfigurationModal.tsx (hunk 2 above)
Comment (knajjars) [reply src/interfaces/assistants_web/src/components/Conversation/Composer/WebSearchConfigurationModal.tsx:27]: this component is not used and is not part of the new designs, will most likely be deleted

**Review by tomtobac — APPROVED** [review]

**Pull request merged**